decidim-assemblies/app/events/decidim/assembly_role_assigned_event.rb

```diff
@@ -0,0 +1,20 @@
+# frozen-string_literal: true
+
+module Decidim
+  class AssemblyRoleAssignedEvent < Decidim::Events::BaseEvent
+    include Decidim::Events::NotificationEvent
+  
+    def notification_title
+      I18n.t("notification_title", i18n_options).html_safe
+    end
+
+    def i18n_options
+      {
+        resource_title: resource_title,
+        resource_url: resource_url,
+        scope: event_name,
+        role: extra["role"]
```

Comment:
We're sending the raw value of the role to the event, and then use that value when showing the i18n string. This will result in an English mashup when using locales other than English. Maybe we could refactor this? I think it's fine sending the raw value, this way we can translate it here:

```ruby
def i18n_role
  I18n.t(extra["role"], "decidim.admin.models.assembly_user_role.roles", default: nil)
end

def i18n_options
  {
    ...
    role: i18n_role
  }
end

def notification_title
  return I18n.t("notification_title", i18n_options).html_safe if i18n_role

  I18n.t("notification_title_general_role", i18n_options).html_safe
end
```

And then add the `notification_title_general_role` with the string `"You have been assigned a role to %{resource_title}"` or something like that. 

What do you think? 😄 

This would need to be applied to all similar classes!

Reply:
Sounds good to me!